src/main/kotlin/io/cloudflight/gradle/autoconfigure/java/JavaConfigurePlugin.kt

```diff
@@ -64,9 +88,20 @@ class JavaConfigurePlugin : Plugin<Project> {
                 it.options.encoding = javaConfigureExtension.encoding.get()
             }
         }
+
+        // TODO pull that out of the JavaConfigurePlugin?
+        project.withPlugin(SpringBootPlugin::class.java) {
+            SpringBootExtension.configure(project)
+        }
+        project.withPlugin(ShadowPlugin::class.java) {
+            ShadowExtension.configure(project)
+        }
+        project.withPlugin(GitPropertiesPlugin::class.java) {
+            GitExtension.configure(project)
+        }
     }
 
-    private object PopulateManifestAction : Action<Task> {
+    internal object PopulateManifestAction : Action<Task> {
```

Comment:
if we use it in multiple places we should move it to its own file and make it a standalone object